### PR TITLE
Downport, add exit, fixes #2671

### LINKED ISSUE #2671: Some error on ABAP 7.02
1 - **CALL FUNCTION 'SEO_INTERFACE_CREATE_COMPLETE'** (line 38282) don't have **suppress_dialog** parameter and dump when run.

2 - **INSERT REPORT iv_program FROM it_source EXTENSION TYPE iv_extension STATE iv_version PROGRAM TYPE iv_program_type.** (line 38358) - dump with error **INSERT_REPORT_NO_EXTTYPE**

### PR DESCRIPTION
**Pull request by sandraros**
Solves the issue #2671

1) Non-existing parameter SUPPRESS_DIALOG in old version of SEO_CLASS_CREATE_COMPLETE
-> Catch exception cx_sy_dyn_call_param_not_found -> call again without SUPPRESS_DIALOG

2) Short dump INSERT_REPORT_NO_EXTTYPE because extension type CS is not supported in old 7.02 kernel.
-> New user exit method CUSTOM_DESERIALIZE_ABAP_CLIF 
-> I also reorganized a little bit the objects
-> Documentation of user exits updated

I also reorganized a little bit the objects:
- methods for deserializing classes are moved from ZCL_ABAPGIT_OO_CLASS to ZCL_ABAPGIT_OO_DESERIALIZER (new class, to complete the existing ZCL_ABAPGIT_OO_SERIALIZER)
- methods for deserializing interfaces are moved down from parent ZCL_ABAPGIT_OO_BASE to subclass ZCL_ABAPGIT_OO_INTERFACE - it was not required but it brings clarity
- code for activation of class is moved down from ZCL_ABAPGIT_OO_CLASS to ZCL_ABAPGIT_OBJECT_CLAS - it was not required but it brings clarity

### CODE REVIEW

File: src/objects/zcl_abapgit_object_clas.clas.abap [hunk 1]
```diff
@@ -330,6 +330,11 @@ CLASS ZCL_ABAPGIT_OBJECT_CLAS IMPLEMENTATION.
                       iv_package = iv_package ).
 
     deserialize_docu( io_xml ).
+
+    zcl_abapgit_objects_activation=>add(
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_clas.clas.abap]: this should not be needed, and not related to the original issue?

File: src/objects/zcl_abapgit_oo_base.clas.abap [hunk 2]
```diff
@@ -12,21 +12,11 @@ CLASS zcl_abapgit_oo_base DEFINITION PUBLIC ABSTRACT.
   PRIVATE SECTION.
     DATA mv_skip_test_classes TYPE abap_bool.
 
-    METHODS deserialize_abap_source_old
-      IMPORTING is_clskey TYPE seoclskey
-                it_source TYPE zif_abapgit_definitions=>ty_string_tt
-      RAISING   zcx_abapgit_exception.
-
-    METHODS deserialize_abap_source_new
```
Comment (larshp) [inline src/objects/zcl_abapgit_oo_base.clas.abap]: this is deleted? but required in the code, is this some separate refactoring task? Its really difficult to review a fix with 1000+ lines changed with unrelated changes

File: src/objects/zcl_abapgit_oo_base.clas.abap [hunk 3]
```diff
@@ -168,15 +71,7 @@ CLASS ZCL_ABAPGIT_OO_BASE IMPLEMENTATION.
...
-    TRY.
-        deserialize_abap_source_new(
-          is_clskey = is_key
-          it_source = it_source ).
-      CATCH cx_sy_dyn_call_error.
-        deserialize_abap_source_old(
-          is_clskey = is_key
-          it_source = it_source ).
-    ENDTRY.
+    ASSERT 0 = 1. "Subclass responsibility
```
Comment (larshp) [inline src/objects/zcl_abapgit_oo_base.clas.abap]: why is this moved?

File: src/objects/zcl_abapgit_oo_deserializer.clas.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<abapGit version="v1.0.0" serializer="LCL_OBJECT_CLAS" serializer_version="v1.0.0">
+ <asx:abap xmlns:asx="http://www.sap.com/abapxml" version="1.0">
+  <asx:values>
```
Comment (larshp) [inline src/objects/zcl_abapgit_oo_deserializer.clas.xml]: is this a new class or something renamed?

File: src/objects/zcl_abapgit_object_clas.clas.abap (hunk 1 above)
Comment (sandraros) [reply src/objects/zcl_abapgit_object_clas.clas.abap]: It's moved from ZCL_ABAPGIT_OO_CLASS, I was influenced by "TODO, perhaps move this call to somewhere else, to be done while cleaning up the CLAS deserialization", but I realize that there's also another such an activation for REPT still in ZCL_ABAPGIT_OO_CLASS. Okay to keep it at its original place.

File: src/objects/zcl_abapgit_oo_base.clas.abap (hunk 2 above)
Comment (sandraros) [reply src/objects/zcl_abapgit_oo_base.clas.abap]: Not deleted: it's moved to the new class ZCL_ABAPGIT_OO_DESERIALIZER. These two methods are used only for interfaces (classes are deserialized in ZCL_ABAPGIT_OO_CLASS), so it should at least be in ZCL_ABAPGIT_OO_INTERFACE. I preferred to move everything to ZCL_ABAPGIT_OO_DESERIALIZER (same design as for the serialization, i.e. everything in class ZCL_ABAPGIT_OO_SERIALIZER). Let me rework that in 2 PRs.

File: src/objects/zcl_abapgit_oo_base.clas.abap (hunk 3 above)
Comment (sandraros) [reply src/objects/zcl_abapgit_oo_base.clas.abap]: Not related to the issue. This code is called only for interfaces, so I think it should be moved at least to subclass ZCL_ABAPGIT_OO_INTERFACE. See my comment in above review.

File: src/objects/zcl_abapgit_oo_deserializer.clas.xml (hunk 4 above)
Comment (sandraros) [reply src/objects/zcl_abapgit_oo_deserializer.clas.xml]: ZCL_ABAPGIT_OO_DESERIALIZER is a new class containing moved methods. Not related to the issue but helps organizing and simplifying for solving the issue. I propose to split in 2 PRs.

**Comment by sandraros** [comment]
Do you prefer that I cancel this PR, and create two PRs, one with the technical moves and groupings, and then one to solve?

**Comment by larshp** [comment]
yes, please split it into smaller bits, this makes it easier to review

**Comment by larshp** [comment]
closing, please split into smaller PRs
